File: 第7章 曲线之美/7.1 笛卡儿心形曲线/笛卡儿心形曲线.py
'''
程序：笛卡儿心形曲线
作者：苏秦@小海豚科学馆公众号
来源：图书《趣味编程三剑客：从Scratch迈向Python和C++》
'''
from turtle import *
from math import pi, sin, cos

def draw():
    '''绘制笛卡儿心形曲线'''
    a, t = 80, 90
    for i in range(360):
        rad = pi / 180 * t                # 角度转为弧度
        x = cos(rad) * a * (1 - sin(rad))
        y = sin(rad) * a * (1 - sin(rad))
        goto(x, y)
        t += 1

if __name__ == '__main__':
    '''画笔初始化'''
    color('red')
    pensize(3)
    draw()

File: 第7章 曲线之美/7.1 笛卡儿心形曲线/笛卡儿心形曲线.cpp
/*
程序：笛卡尔心形曲线
作者：苏秦@小海豚科学馆公众号
来源：图书《趣味编程三剑客：从Scratch迈向Python和C++》
*/
// 绘制笛卡尔心形曲线
void draw()
{
	const float PI = 3.14159;
	int a = 80, t = 90;
	for (int i = 0; i < 360; i++) {
		float rad = PI / 180 * t;                // 角度转为弧度
        float x = cos(rad) * a * (1 - sin(rad));
        float y = sin(rad) * a * (1 - sin(rad));
        pen.line(x, y);
        t += 1;
    }
}

int main()
{
	// 画笔初始化
    pen.color(_red);
    pen.size(3);
	draw();
    return 0;
}
